x/vm/types/tx.go

```diff
@@ -25,24 +26,33 @@ type EvmTxArgs struct {
 }
 
 // ToTxData converts the EvmTxArgs to TxData
-func (args *EvmTxArgs) ToTxData() (TxData, error) {
-	ethTx := NewTx(args).AsTransaction()
-	return NewTxDataFromTx(ethTx)
+func (args *EvmTxArgs) ToTxData() *TransactionArgs {
+	return &TransactionArgs{
+		Nonce:                (*hexutil.Uint64)(&args.Nonce),
+		Gas:                  (*hexutil.Uint64)(&args.GasLimit),
+		Data:                 (*hexutil.Bytes)(&args.Input),
+		MaxFeePerGas:         (*hexutil.Big)(args.GasFeeCap),
+		GasPrice:             (*hexutil.Big)(args.GasPrice),
+		ChainID:              (*hexutil.Big)(args.ChainID),
+		MaxPriorityFeePerGas: (*hexutil.Big)(args.GasTipCap),
+		Value:                (*hexutil.Big)(args.Amount),
+		To:                   args.To,
+		AccessList:           args.Accesses,
```

Comment:
```suggestion
		AccessList:           args.Accesses,
		AuthorizationList:    args.Authrozations,
```

I think you need to add an `AuthorizationList` for the `SetCodeTx` type.